src/Adapter/MSTest.CoreAdapter/Helpers/ReflectHelper.cs

```diff
@@ -130,6 +130,43 @@ public bool HasAttributeDerivedFrom(MemberInfo memberInfo, Type baseAttributeTyp
             return false;
         }
 
+        /// <summary>
+        /// Returns true when specified class/member has attribute which implements specific interface.
+        /// </summary>
+        /// <param name="memberInfo">The member info.</param>
+        /// <param name="interfaceType">The interface type.</param>
+        /// <param name="inherit">Should look at inheritance tree.</param>
+        /// <returns>An object derived from Attribute that corresponds to the instance of found attribute.</returns>
+        public virtual bool DoesAttributeImplement(MemberInfo memberInfo, Type interfaceType, bool inherit)
+        {
+            if (memberInfo == null)
+            {
+                throw new ArgumentNullException(nameof(memberInfo));
+            }
+
+            if (interfaceType == null)
+            {
+                throw new ArgumentNullException(nameof(interfaceType));
+            }
+
+            // Get all attributes on the member.
+            Dictionary<string, object> attributes = this.GetAttributes(memberInfo, inherit);
+
+            // Try to find the attribute that is derived from baseAttrType.
```

Comment:
there is no baseAttrType in this context

Reply:
Done